src/ansys/mechanical/core/embedding/app.py

```diff
@@ -194,9 +195,43 @@ def save(self, path=None):
         else:
             self.DataModel.Project.Save()
 
-    def save_as(self, path):
-        """Save the project as."""
-        self.DataModel.Project.SaveAs(path)
+    def save_as(self, path: str, overwrite: bool = False):
+        """Save the project as a new file.
+
+        If the `overwrite` flag is enabled, the current saved file is temporarily moved
+        to a backup location. The new file is then saved in its place. If the process fails,
+        the backup file is restored to its original location.
+
+        Parameters
+        ----------
+        path: int, optional
+            The path where file needs to be saved.
+        overwrite: bool, optional
+            Whether the file should be overwritten if it already exists (default is False).
+        """
+        if not os.path.exists(path):
+            self.DataModel.Project.SaveAs(path)
+            return
+        else:
+            if not overwrite:
+                raise Exception(
+                    f"File already exists in {path}, Use ``overwrite`` flag to "
+                    "replace the existing file."
+                )
+
+            file_name = os.path.basename(path)
+            file_dir = os.path.dirname(path)
+            associated_dir = os.path.join(file_dir, os.path.splitext(file_name)[0] + "_Mech_Files")
+
+            try:
+                # Remove existing files and associated folder
+                os.remove(path)
+                if os.path.exists(associated_dir):
+                    shutil.rmtree(associated_dir)
+                # Save the new file
+                self.DataModel.Project.SaveAs(path)
+            except Exception as e:
+                raise e
```

Comment:
why a try/except just to rethrow the exception? Is this just for debugging?

Reply:
Was part of previous method of restoring backup. I will update it.